Userland/Libraries/LibWeb/TreeNode.h

```diff
@@ -153,6 +171,28 @@ class TreeNode : public Weakable<T> {
         return false;
     }
 
+    // https://dom.spec.whatwg.org/#concept-tree-preceding (Object A is 'typename U' and Object B is 'this')
+    template<typename U>
+    bool has_preceding_node_of_type_in_tree_order() const
+    {
+        for (auto* node = previous_in_pre_order(); node; node->previous_in_pre_order()) {
```

Comment:
This loop looks wrong
```suggestion
        for (auto* node = previous_in_pre_order(); node; node = node->previous_in_pre_order()) {
```

Reply:
Oh oops..fixed.